lib/src/Pos/Generator/Block/Logic.hs

```diff
@@ -80,12 +82,14 @@ genBlock eos = do
         EpochOrSlot (Right slot@SlotId {..}) -> withCurrentSlot slot $ do
             genPayload slot
             -- We need to know leader's secret key to create a block.
-            leader <-
+            precomputedLeader <-
```

Comment:
Well, I don't know why you renamed it (leader is leader even if she delegated), but it's not essential.